org.opentosca.planbuilder/org.opentosca.planbuilder.type.plugin.connectsto/src/main/java/org/opentosca/planbuilder/type/plugin/connectsto/bpel/handler/BPELConnectsToPluginHandler.java

```diff
@@ -126,6 +113,37 @@ private boolean executeConnectsTo(final BPELPlanContext templateContext, final T
                         BPELConnectsToPluginHandler.LOG.warn("Didn't find necessary matchings from parameter to property. Can't initialize connectsTo relationship.");
                     } else {
                         // executable operation found
+                        BPELConnectsToPluginHandler.LOG.debug("Source: " + sourceParameterNode.getName());
+                        BPELConnectsToPluginHandler.LOG.debug("Target: " + targetParameterNode.getName());
+                        BPELConnectsToPluginHandler.LOG.debug(iface.getName());
+                        if (iface.getName().matches(".*/source(/[^/]+)?$")) {
+                            BPELConnectsToPluginHandler.LOG.debug("connectTo source defined");
+                            if (!connectToNode.equals(sourceParameterNode)) {
+                                BPELConnectsToPluginHandler.LOG.debug("source does not match");
+                                continue;
+                            }
+                            if (!iface.getName().endsWith("/source")) {
+                                final String targetTypeWithoutVersion = targetParameterNode.getType().getLocalPart().split("_")[0];
```

Comment:
@lharzenetter is there a better way to check only for the nodetype id/name ? Via WineryVersionUtils and cutting the parts out from the back ?